File: go-shell/internal/resources/resources.go
package resources

import (
	"encoding/json"
	"os"

	toml "github.com/pelletier/go-toml/v2"
	"gopkg.in/yaml.v3"
)

// LoadJSON reads a JSON file from disk.
func LoadJSON(path string) (map[string]any, error) {
	b, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	var v map[string]any
	if err := json.Unmarshal(b, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// LoadYAML reads a YAML file from disk.
func LoadYAML(path string) (map[string]any, error) {
	b, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	var v map[string]any
	if err := yaml.Unmarshal(b, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// LoadTOML reads a TOML file from disk.
func LoadTOML(path string) (map[string]any, error) {
	b, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	var v map[string]any
	if err := toml.Unmarshal(b, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// LoadPrompt reads a plain text prompt template.
func LoadPrompt(path string) (string, error) {
	b, err := os.ReadFile(path)
	return string(b), err
}

File: aider-cli/src/resources.rs
use std::fs;
use std::path::Path;

use anyhow::Result;
use serde_json::Value as JsonValue;
use serde_yaml::Value as YamlValue;
use toml::Value as TomlValue;

/// Load a JSON resource from disk.
pub fn load_json<P: AsRef<Path>>(path: P) -> Result<JsonValue> {
    let data = fs::read_to_string(path)?;
    Ok(serde_json::from_str(&data)?)
}

/// Load a YAML resource from disk.
pub fn load_yaml<P: AsRef<Path>>(path: P) -> Result<YamlValue> {
    let data = fs::read_to_string(path)?;
    Ok(serde_yaml::from_str(&data)?)
}

/// Load a TOML resource from disk.
#[allow(dead_code)]
pub fn load_toml<P: AsRef<Path>>(path: P) -> Result<TomlValue> {
    let data = fs::read_to_string(path)?;
    Ok(toml::from_str(&data)?)
}

/// Load a plain text prompt template.
pub fn load_prompt<P: AsRef<Path>>(path: P) -> Result<String> {
    Ok(fs::read_to_string(path)?)
}
